src/clients/c++/experimental_api_v2/library/common.h

```diff
@@ -0,0 +1,250 @@
+  /// \param dims the vector of dims representing the new shape
+  /// of input.
+  /// \return Error object indicating success or failure of the
+  /// request.
+  virtual Error SetShape(const std::vector<int64_t>& dims) = 0;
+
+  /// Set the tensor data from the specified buffer described as
+  /// starting address and buffer size.
+  /// \param input The base pointer of buffer holding the input data.
+  /// \param input_byte_size The size of buffer in bytes.
+  /// \return Error object indicating success or failure of the
+  /// request.
+  virtual Error SetRaw(const uint8_t* input, size_t input_byte_size) = 0;
+
+  /// Set the tensor data from the specified buffer represented
+  /// as vector.
+  /// \param input The vector holding input data.
+  /// \return Error object indicating success or failure of the
+  /// request.
+  virtual Error SetRaw(const std::vector<uint8_t>& input) = 0;
+
+  /// Set the tensor data from the specified vector of strings, where
+  /// each string represents an lement of the "BYTES" tensor ordered
+  /// in row-major format.
+  /// \param input The vector holding strings.
+  /// \return Error object indicating success or failure of the
+  /// request.
+  virtual Error SetFromString(const std::vector<std::string>& input) = 0;
+
+  /// Set the tensor data to be read from a registered shared memory region.
+  /// \param region_name The name of the shared memory region.
+  /// \param byte_size The size of data in bytes.
+  /// \param offset The offset in shared memory region. Default value is 0.
+  /// \return Error object indicating success or failure of the
+  /// request.
+  virtual Error SetSharedMemory(
+      const std::string& region_name, const size_t byte_size,
+      const size_t offset = 0) = 0;
+};
+
+//==============================================================================
+/// An interface for InferOutput object to describe the requested model
+/// output for inference.
+///
+class InferOutput {
+ public:
+  /// Get the name of output associated with this object
+  /// \param name Returns the name of output.
+  /// \return Error object indicating success or failure of the
+  /// request.
+  virtual Error GetName(std::string* name) const = 0;
+
+  /// Set the output tensor data to be written to specified shared
+  /// memory region.
+  /// \param region_name The name of the shared memory region.
+  /// \param byte_size The size of data in bytes.
+  /// \param offset The offset in shared memory region. Default value is 0.
+  /// \return Error object indicating success or failure of the
+  /// request.
+  virtual Error SetSharedMemory(
+      const std::string& region_name, const size_t byte_size,
+      const size_t offset = 0) = 0;
+};
+
+//==============================================================================
+/// An interface for InferResult object which allows to access and
+/// interpret the response from inference request.
+///
+class InferResult {
```

Comment:
InferOutput? And in the description make sure to emphasis that this represents an output tensor in the response. As it is now is sounds like it is the entire response.

Reply:
InferResult object does point to the entire response. The users while obtaining the data for a particular output has to provide the output name and the pointer to the buffer in the message will be returned. It is similar to our python side.